relay-server/src/actors/processor.rs

```diff
@@ -1352,6 +1363,19 @@ impl EnvelopeProcessorService {
         //  2. The DSN was moved and the envelope sent to the old project ID.
         envelope.meta_mut().set_project_id(project_id);
 
+        let reservoir = {
+            #[allow(unused_mut)]
+            let mut reservoir = ReservoirEvaluator::new(reservoir_counters);
+
+            #[cfg(feature = "processing")]
+            if let Some(redis_pool) = self.inner.redis_pool.as_ref() {
+                let org_id = managed_envelope.scoping().organization_id;
+                reservoir = reservoir.set_redis(org_id, redis_pool.clone());
+            }
+
+            Arc::new(reservoir)
```

Comment:
This `Arc` is not strictly necessary. The `SamplingEvaluator` can hold a mutable reference to the `ReservoirEvaluator`, which means you can pass `&mut state.reservoir` into `compute_sampling_decision`.